toolbox/library/src/YarpRpc.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include "WBToolbox/Block/YarpRpc.h"
+
+#include <BlockFactory/Core/BlockInformation.h>
+#include <BlockFactory/Core/Log.h>
+#include <BlockFactory/Core/Parameter.h>
+#include <BlockFactory/Core/Parameters.h>
+#include <BlockFactory/Core/Signal.h>
+
+#include <yarp/os/Network.h>
+#include <yarp/os/RpcClient.h>
+#include <yarp/os/Bottle.h>
+
+#include <string>
+
+using namespace wbt::block;
+using namespace blockfactory::core;
+
+// INDICES: PARAMETERS, INPUTS, OUTPUT
+// ===================================
+
+enum ParamIndex
+{
+    Bias = Block::NumberOfParameters - 1,
+    PortName,
+    RpcCommand,
+};
+
+enum InputIndex
+{
+    TriggerSignal = 0,
+};
+
+// BLOCK PIMPL
+// ===========
+
+class YarpRpc::impl
+{
+public:
+    std::string serverPortName;
+    std::string clientPortName;
+
+    std::unique_ptr<yarp::os::Network> network = nullptr;
+    yarp::os::RpcClient rpcClientPort;
+    yarp::os::Bottle rpcCommand;
+};
+
+// BLOCK CLASS
+// ===========
+
+YarpRpc::YarpRpc()
+    : pImpl{new impl()}
+{}
+
+YarpRpc::~YarpRpc() = default;
+
+unsigned YarpRpc::numberOfParameters()
+{
+    return Block::numberOfParameters() + 2;
+}
+
+bool YarpRpc::parseParameters(BlockInformation* blockInfo)
+{
+    const std::vector<ParameterMetadata> metadata{
+        {ParameterType::STRING, ParamIndex::PortName, 1, 1, "PortName"},
+        {ParameterType::STRING, ParamIndex::RpcCommand, 1, 1, "RpcCommand"}
+    };
+
+    for (const auto& md : metadata) {
+        if (!blockInfo->addParameterMetadata(md)) {
+            bfError << "Failed to store parameter metadata";
+            return false;
+        }
+    }
+
+    return blockInfo->parseParameters(m_parameters);
+}
+
+bool YarpRpc::configureSizeAndPorts(BlockInformation* blockInfo)
+{
+    // PARAMETERS
+    // ==========
+
+    if (!YarpRpc::parseParameters(blockInfo)) {
+        bfError << "Failed to parse parameters.";
+        return false;
+    }
```

Comment:
This is required only if in this method you need some information from the parameters. Since this is not the case, you can remove this `if` statement.

Reply:
Done